src/utils/regexp.ts

```diff
@@ -1,5 +1,33 @@
+import { Context } from "../api";
 import { assert } from "./errors";
 
+
+/**
+ * Wrapper around RegExp.exec that takes into account the user's smart case
+ * preference.
+ */
+export function smartExec(re: RegExp, string: string): RegExpExecArray | null {
```

Comment:
By convention you should accept a `Context` as argument (and assign it to `Context.current`). In this case you can event take a `Context.WithoutActiveEditor`, and also behave correctly when there is no active context for some reason:

```suggestion
export function smartExec(
  re: RegExp,
  string: string,
  context = Context.WithoutActiveEditor.currentOrUndefined,
): RegExpExecArray | null {
```

That way callers can avoid a call to `.currentOrUndefined` by passing in a context directly.